staging/src/k8s.io/apimachinery/pkg/util/net/interface.go

```diff
@@ -110,36 +113,41 @@ func isLoopbackOrPointToPoint(intf *net.Interface) bool {
 	return false
 }
 
-//getFinalIP method receives all the IP addrs of a Interface
-//and returns a nil if the address is Loopback, Ipv6, link-local or nil.
-//It returns a valid IPv4 if an Ipv4 address is found in the array.
-func getFinalIP(addrs []net.Addr) (net.IP, error) {
+func inFamily(ip net.IP, expectedFamily AddressFamily) bool {
+	ipFamily := familyIPv4
+	if ip.To4() == nil {
+		ipFamily = familyIPv6
+	}
+	return ipFamily == expectedFamily
+}
+
+// getFinalIP method checks all the IP addresses of a Interface looking
```

Comment:
The function seems to return the first valid global scope IP, not the "final" IP. Can we rename this to "GetFirstGlobalIP" or something similar?

Also, as commented on the previous patch, I think it's better to return a list of global IPs and then have the caller explicitly pick off the first element of the list...